rpc/server.go

```diff
@@ -210,55 +212,40 @@ func marshalResponse(response any, log *zerolog.Logger) []byte {
 	return responseData
 }
 
-type jsonrpcReq struct {
-	Error  []byte
-	Method string
-	Id     uint64
-}
-
-func validateJsonrpcRequest(requestData []byte, logger *zerolog.Logger) jsonrpcReq {
+func validateJsonrpcRequest(requestData []byte, logger *zerolog.Logger) (types.JsonRpcRequest, []byte) {
 	var request map[string]interface{}
-	vr := jsonrpcReq{}
+	vr := types.JsonRpcRequest{}
 	err := json.Unmarshal(requestData, &request)
 	if err != nil {
-		vr.Error = marshalResponse(types.UnexpectedRequestUnmarshalError, logger)
-		return vr
+		errRes := types.JsonRpcResponse{Jsonrpc: "2.0", ErrorObj: types.UnexpectedRequestUnmarshalError}
+		return types.JsonRpcRequest{}, marshalResponse(errRes, logger)
 	}
 
 	// jsonrpc spec says id can be a string, number.
 	// We only support numbers: https://github.com/statechannels/go-nitro/issues/1160
 	// When golang unmarshals JSON into an interface value, float64 is used for numbers.
 	requestId := request["id"]
 	fRequestId, ok := requestId.(float64)
-	if !ok {
-		vr.Error = marshalResponse(types.InvalidRequestError, logger)
-		return vr
-	}
-
-	if fRequestId != float64(uint64(fRequestId)) {
-		vr.Error = marshalResponse(types.InvalidRequestError, logger)
-		return vr
+	if !ok || fRequestId != float64(uint64(fRequestId)) {
+		errRes := types.JsonRpcResponse{Jsonrpc: "2.0", ErrorObj: types.InvalidRequestError}
+		return types.JsonRpcRequest{}, marshalResponse(errRes, logger)
 	}
 	vr.Id = uint64(fRequestId)
 
 	sJsonrpc, ok := request["jsonrpc"].(string)
 	if !ok || sJsonrpc != "2.0" {
-		requestError := types.InvalidRequestError
-		requestError.Id = vr.Id
-		vr.Error = marshalResponse(requestError, logger)
-		return vr
+		errRes := types.JsonRpcResponse{Jsonrpc: "2.0", Id: vr.Id, ErrorObj: types.InvalidRequestError}
+		return types.JsonRpcRequest{}, marshalResponse(errRes, logger)
 	}
 
-	_, ok = request["method"].(string)
+	sMethod, ok := request["method"].(string)
 	if !ok {
-		requestError := types.InvalidRequestError
-		requestError.Id = vr.Id
-		vr.Error = marshalResponse(requestError, logger)
-		return vr
+		errRes := types.JsonRpcResponse{Jsonrpc: "2.0", Id: vr.Id, ErrorObj: types.InvalidRequestError}
```

Comment:
Nit: We have a pattern elsewhere of using a helper function to construct these objects, rather than doing it literally. Doing that here would reduce the need to have `"Jsonrpc:"2.0"` everywhere, and make future upgrades easier to do? Don't feel strongly though